localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -784,15 +788,53 @@ geometry_msgs::msg::PoseWithCovarianceStamped NDTScanMatcher::align_using_monte_
 
   output_pose_with_cov_to_log(get_logger(), "align_using_monte_carlo_input", initial_pose_with_cov);
 
-  // generateParticle
-  const auto initial_poses =
-    create_random_pose_array(initial_pose_with_cov, initial_estimate_particles_num_);
+  const double CONVERT_COEFF = std::sqrt(2);
+
+  const auto base_rpy = get_rpy(initial_pose_with_cov);
+  const Eigen::Map<const RowMatrixXd> covariance = {
+    initial_pose_with_cov.pose.covariance.data(), 6, 6};
+  const double stddev_x = CONVERT_COEFF * std::sqrt(covariance(0, 0));
+  const double stddev_y = CONVERT_COEFF * std::sqrt(covariance(1, 1));
+  const double stddev_z = CONVERT_COEFF * std::sqrt(covariance(2, 2));
+  const double stddev_roll = CONVERT_COEFF * std::sqrt(covariance(3, 3));
+  const double stddev_pitch = CONVERT_COEFF * std::sqrt(covariance(4, 4));
+
+  auto uniform_to_normal = [](const double uniform) {
+    assert(-1.0 <= uniform && uniform <= 1.0);
+    constexpr double epsilon = 1.0e-6;
+    const double clamped = std::clamp(uniform, -1.0 + epsilon, 1.0 - epsilon);
+    return boost::math::erf_inv(clamped);
+  };
+
+  auto normal_to_uniform = [](const double normal) { return boost::math::erf(normal); };
```

Comment:
These conversions are difficult to understand. It is not intuitive why the conversion coefficient is the SQRT(2) and why the conversions between uniform and normal distributions are necessary.

Is it possible to perform these conversions in the `TreeStructuredParzenEstimator` class?
For example, adding modes for normal distribution and uniform distribution to the class.

Reply:
Added a comment and changed the code slightly.
I am still considering whether to change the structure.
https://github.com/autowarefoundation/autoware.universe/pull/5078/commits/00b0adc3c7935780afa69483f0d734fad458687a